tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -803,9 +803,12 @@ GDScript:
           in the shader. It must match *exactly* to the name of the uniform in
           the shader or else it will not be recognized.
 
-Any GLSL type except for *void* can be a uniform. Additionally, Godot provides
-optional shader hints to make the compiler understand for what the uniform is
-used, and how the editor should allow users to modify it.
+
+Uniform hints
+~~~~~~~~~~~~~
+
+Additionally, Godot provides optional uniform hints to make the compiler understand
+what the uniform is used for, and how the editor should allow users to modify it.
```

Comment:
```suggestion
Godot provides optional uniform hints to make the compiler understand
what the uniform is used for, and how the editor should allow users to modify it.
```
Since this is the start of a new section and paragraph